crates/trie/common/src/proofs.rs

```diff
@@ -548,6 +548,46 @@ impl AccountProof {
                 .collect(),
         }
     }
+
+    /// Converts an
+    /// [`EIP1186AccountProofResponse`](alloy_rpc_types_eth::EIP1186AccountProofResponse) to an
+    /// [`AccountProof`].
+    ///
+    /// This is the inverse of [`Self::into_eip1186_response`]
+    pub fn from_eip1186_proof(proof: alloy_rpc_types_eth::EIP1186AccountProofResponse) -> Self {
+        let alloy_rpc_types_eth::EIP1186AccountProofResponse {
+            nonce,
+            address,
+            balance,
+            code_hash,
+            storage_hash,
+            account_proof,
+            storage_proof,
+            ..
+        } = proof;
+        let storage_proofs = storage_proof.into_iter().map(Into::into).collect();
+
+        let (storage_root, info) = if nonce == 0 &&
+            balance.is_zero() &&
+            storage_hash.is_zero() &&
+            code_hash == KECCAK_EMPTY
```

Comment:
@Rjected I think this was actually incorrect previously (.is_zero()) we should check for keccak here I believe

Reply:
Empty root for storage hash yeah

Reply:
Would be a good method on the proof object itself I think